openhcl/virt_mshv_vtl/src/processor/mod.rs

```diff
@@ -1166,11 +1203,17 @@ fn from_seg(reg: HvX64SegmentRegister) -> x86defs::SegmentRegister {
     }
 }
 
-struct UhEmulationState<'a, 'b, T: CpuIo, U: Backing> {
+struct UhEmulationState<'a, 'b, T: CpuIo, U: Backing, S: UhX86EmulatorRegisters<'a, 'b, U>> {
     vp: &'a mut UhProcessor<'b, U>,
     interruption_pending: bool,
     devices: &'a T,
     vtl: GuestVtl,
+    registers: S,
+}
+
+pub trait UhX86EmulatorRegisters<'a, 'b, U: Backing> {
```

Comment:
This trait doesn't really seem like it provides all that much value. Can we get rid of it if we go with my idea of hanging the register type off of the backing?

Reply:
Which i think would then require moving flush to the X86EmulatorSupport trait instead, but then it wouldn't need a vp and vtl passed in, it'd already have them from the UhEmulationState.

Reply:
Fixed!